source/server/active_tcp_listener.h

```diff
@@ -7,17 +7,18 @@
 #include "source/common/common/linked_object.h"
 #include "source/common/stream_info/stream_info_impl.h"
 #include "source/server/active_listener_base.h"
+#include "source/server/active_tcp_socket.h"
 
 namespace Envoy {
 namespace Server {
 
 struct ActiveTcpConnection;
 using ActiveTcpConnectionPtr = std::unique_ptr<ActiveTcpConnection>;
-struct ActiveTcpSocket;
-using ActiveTcpSocketPtr = std::unique_ptr<ActiveTcpSocket>;
 class ActiveConnections;
 using ActiveConnectionsPtr = std::unique_ptr<ActiveConnections>;
```

Comment:
Maybe rename the ActiveConnectionsPtr and ActiveConnections to ActiveTcpConnectionsPtr and ActiveTcpConnections, since you define same alias at https://github.com/envoyproxy/envoy/pull/16947/files#diff-7f9735a83df97d06bc763ab41de1eb15787011587b0d18766c1d8848cb8861c0R171, it it a little confuse.

Reply:
I like it.
@antoniovicente  WDYT?

Reply:
We still have the issue of TypedActiveStreamListenerBase and different collections.  I would prefer that the two collections share a base class and eliminate the need for template base classes like this one.

Reply:
ping